Readme.md
# FvwmWeb (fvwm.org) at GitHub

This is the source to build the Fvwm Website: <http://fvwm.org/>

The Wiki is built at: <http://fvwm.org/Wiki/>. This info is the old
wiki info that mostly applies to the full WebSite now. **Updates to come.**

The wiki is built from Markdown files with front matter using Jekyll:
<https://jekyllrb.com>.

To contribute to the wiki you will need to submit pull requests against
this source.

## Layout and info for contributors

The wiki is currently divided into the following main pages.

| Page | Type |
|------|------|
| / | main |
| /Config | config |
| /Config/Functions | functions |
| /CookBook | recipe |
| /Tips | tip |
| /Modules | module |
| /Decor | decor |
| /Panels | panel |
| /Menus | menu |
| /irc | irc |

Each page above generates a list of links to pages based on their type. To create
a new page in one of the above lists first create a directory and index.md
file for your page. For example if you wanted to put a new CookBook recipe,
MyRecipe, you would create the following file

    /Wiki/CookBook/MyRecipe/index.md

Put the following front matter at the top of your index.md file

    ---
    layout : wiki
    title : MyRecipe
    type : recipe
    weight : 250
    description : |
      This is a multiline
      description of MyRecipe.
    ---

This sets the variables that are used when building. The layout must
be set to "wiki" and the title is the title of your page. type, weight and
description are used when building the lists on the main pages.

Set type to the main page you want a link on. The weight is used
to order the list. 1000 is the current top of the list and steps are by 50
or 100 to give room to put pages in between. The description is what appears
below the link when describing that page in the list.

Under the front matter you then write your page using markdown syntax. Here
are some notes.

+ To include a table of contents built from the markdown headers use

        * TOC
        {:toc}

  This will float the TOC to the right. To use a full width TOC use

      * TOC
      {:toc #toc-full}  

+ __Note:__ Syntax highlighting currently doesn't work on github. This
  is here for the possibility of adding it back (see below). The below
  should still be used for fvwm configuration code.

  To use syntax highlighting you need to put your configuration code between
  special code block as follows

      {% highlight fvwm %}
      # Example Function
      DestroyFunc FuncName
      AddToFunc FuncName
      + ...
      {% endhighlight %}

  Non-highlighted code blocks can be generated by indenting them four spaces.

+ For links between pages please use the following (so they can be built for
  different locations):

      [/CookBook/MyRecipe]({{ "/CookBook/MyRecipe" | prepend: site.wikiurl }})

  This uses the variable wikibaseurl from the \_config.yml file to expand the url
  in the link.

+ Include any images or extra files for your page in the directory alongside the
  index.md file.

+ If you want to include links in your description you must write them using html.

        <a href="{{ "/CookBook/MyRecipe" | prepend: site.wikiurl }}">/CookBook/MyRecipe</a>

  This is because the variable is not parsed through the markdown processor.

+ If you want to generate your own main page, look at the current ones for examples
  on building a list from these variables. You need to use an index.html file to
  generate lists you want to be able to use links from the description variable.

## Page Naming Convention

To follow the fvwm config syntax, the naming convention of wiki pages is to
capitalize the first letter of each word. For example the wiki page
for the RightPanel is located at [/Wiki/Panels/RightPanel](
http://www.fvwm.org/Wiki/Panels/RightPanel/).

## Building the site with Jekyll

This static site is built from the Markdown files using [Jekyll](
(https://jekyllrb.com/). To build the wiki you need a copy of
the source and Jekyll installed. In which case the wiki can be
built using

    fvwmorg.github.io# jekyll build

You can build and run a local server of the wiki with

    fvwmorg.github.io# jekyll serve

The static .html for the Wiki are built in fvwmorg.github.io/\_site/Wiki.
If using `jekyll serve`, you will be given a local address to view the Wiki
(along with the full fvwm.org site) at, and any changes you made will be
built and updated to be viewed.

## Syntax Highlighting

My past attempts at syntax highlighting can be found on my old unused
wiki sources at <https://github.com/somiaj/fvwmwiki>. These are not
compatible with github pages, so the current Wiki has no syntax highlighting.

It might be possible to add syntax highlighting in the future if someone
were to write a full lexer for [rouge](https://github.com/rouge-ruby/rouge),
and get it accepted into their source. My initial attempts at this can be
found at my outdated fork <https://github.com/somiaj/rouge>.




ChangeLog
2016-03-27 Jaimos Skriletz <jaimosskriletz@boisestate.edu>

	* *: Rebuilt webpage to move to GitHub.
	New webpages are generated by Jekyll.

2015-12-17  Dan Espen  <despen@verizon.net>

	* index.php: Remove "Development is active".

	* download/index.php: Remove reference to snapshots, they haven't been built for ages.
	Change label from "Latest CVS sources via ftp:"
	to "Latest released CVS sources via ftp:".

2014-04-09  Thomas Adam <thomas@fvwm.org>
	* documentation/dev_cvs.php:
	Fix a nasty typo, rendering the <php block useless

2014-04-08  Dan.Espen  <despen@verizon.net>

	* documentation/dev_cvs.php: Fix "<pre class=" in Adding Directories and Files.
	Add example for checking out fvwm-web.
	Replace testing instructions,
	the old "file:" trick won't work since we're all PHP.


2013-04-15  Thomas Adam <thomas@xteddy.org>
	* documentation/dev_cvs.php:
	Remove redundant CVS-specific information.

2013-01-05  Dan Espen  <despen@verizon.net>

	* documentation/faq/faq2php: Add emacs compile-command that works on my setup.
	* index.php: Regenerate for Wine issue.

2012-02-01  Thomas Adam <thomas@fvwm.org>
	* authors/index.php:
	* news/index.php:
	* doc/stable:
	* index.php:
	* documentation/manpages/stable:
	Regenerated for 2.6.5 release.

2012-02-09  Thomas Adam <thomas@fvwm.org>
	* index.php:
	Fix syntax.

2012-02-02  Thomas Adam <thomas@fvwm.org>
	* index.php:
	Auto-generate NEWS section.

2012-02-01  Thomas Adam <thomas@fvwm.org>
	* authors/index.php:
	* news/index.php:
	* doc/stable:
	* index.php:
	* documentation/manpages/stable:
	Regenerated for 2.6.4 releae.

2011-12-10  Thomas Adam <thomas@fvwm.org>
	* screenshots/desktops/Axelle-Apvrille-1024x640:
	Added screenshot.

2011-10-26  Thomas Adam <thomas@fvwm.org>
	* freenode.txt:
	Removed now that group verification complete.

2011-10-25  Thomas Adam <thomas@fvwm.org>
	* freenode.txt:
	Added Freenode verification file for group registrations

	"As part of the verification process for your GRF-f request, please
	upload a static plain-text file to http://www.fvwm.org/freenode.txt with
	the contents "aoisenfto3ib45i3ub&T*T&", and let me know when you've done
	so."

2011-10-07  Thomas Adam <thomas@fvwm.org>
	* links.php:
	Corrected link to zensites.

2011-09-30  Thomas Adam <thomas@fvwm.org>
	* authors/index.php:
	* news/index.php:
	* doc/stable:
	* index.php:
	* documentation/manpages/stable:
	Regenerated for 2.6.3 releae.

2011-08-06  Thomas Adam <thomas@fvwm.org>
	* authors/index.php:
	* news/index.php:
	* doc/stable:
	* documentation/manpages/stable:
	Regenerated for 2.6.2 releae.

2011-07-04  Thomas Adam <thomas@fvwm.org>
	* screenshots/desktops/Helmuth_Schmelzer-1366x768:
	Added screenshot.

2011-06-04  Thomas Adam <thomas@fvwm.org>
	* index.php:
	* download/index.php:
	* definitions.inc:
	Remove references to an unstable FVWM version

	This concept no longer exists in FVWM.

2011-05-15  Thomas Adam <thomas@fvwm.org>
	* download/index.php:
	Deprecate 2.4.X officially now that 2.6.X has been out long enough for
	most packagers to migrate.   Also, mention rather loudly for those
	people on OpenBSD, that the version of FVWM which ships with it, is not
	supported at all.  No matter how hard they might try and argue
	otherwise.

2011-05-15  Thomas Adam <thomas@fvwm.org>
	* download/index.php:
	Remove any mention of RPM/DEB packages.  They've not been updated  in
	ages, and doing so is pointless given downstream's packaging techniques.

2011-05-09  Thomas Adam <thomas@fvwm.org>
	* screenshots/desktops/Susanne_Schmidt-fvwmbrown-1000x750:
	* screenshots/desktops/Susanne_Schmidt-shotgreen-900x889:
	Clarified description; "patched FVWM" needed.

2011-05-05  Thomas Adam <thomas@fvwm.org>
	* screenshots/desktops/Susanne_Schmidt-fvwmbrown-1000x750:
	* screenshots/desktops/Susanne_Schmidt-shotgreen-900x889:
	Added screenshots from Susanne.

2011-04-17  Thomas Adam <thomas@fvwm.org>
	* download/index.php:
	Remove reference to bug tracker.

2011-04-16  Thomas Adam <thomas@fvwm.org>
	* documentation/manpages/stable:
	* index.php:
	* definitions.inc:
	Regenerate files for 2.6.1

2011-04-15  Thomas Adam <thomas@fvwm.org>
	* documentation/manpages/stable:
	Regenerate against a stable source tree, and not unstable.

	This corrects any version numbers on the generated files.

2011-04-15  Thomas Ada, <thomas@fvwm.org>
	* screenshots/desktops/screenshot_db.inc:
	* screenshots/desktops/Christine-blackandgrey-1600x1200:
	* screenshots/desktops/Christine-fukation-1400x1050:
	* screenshots/desktops/Christine-glassy-1024x768:
	* screenshots/desktops/Christine-international-1400x1050:
	* screenshots/desktops/Christine-toogrey-1024x768:
	* screenshots/desktops/Christine-untitled-1600x1200:
	Added screenshots from Christine.

2011-04-15  Thomas Adam <thomas@fvwm.org>
	* documentation/manpages/manpages2php:
	* documentation/manpages/stable:
	* documentation/manpages/unstable:
	* documentation/manpages/index.php:
	Update various files, and regenerate .php files now that 2.6.0 is stable
	as there's a lot of hard-coded assumptions on 2.4.X still.

2011-04-15  Thomas Adam <thomas@fvwm.org>
	* news/index.php:
	* doc/unstable:
	* documentation/manpages/unstable:
	Regenerated for 2.6.0/2.7.0 release.

2011-01-29  Thomas Adam <thomas@fvwm.org>
	* documentation/faq/index.php:
	Updated to reflect addition of #3.30

2010-10-08  Thomas Adam <thomas@fvwm.org>
	* documentation/manpages/unstable:
	Regenerated to remove erroneous link.

2010-10-08  Thomas Adam <thomas@fvwm.org>
	* documentation/manpages/unstable:
	Regenerated.

2010-08-31  Thomas Adam <thomas@xteddy.org>
	* links.php:
	* screenshots/desktops/screenshot_db.inc:
	* screenshots/desktops/Andrew_Dolgov-1152x864/description.inc:
	* screenshots/desktops/Nuno_Alexandre-1600x1200/description.inc:
	* screenshots/desktops/Piotr_Zielinski-1024x768/description.inc:
	* screenshots/desktops/Remko_Troncon-desk-1024x768/description.inc:
	* screenshots/desktops/Thomas_Adam-desk-1280x960/description.inc:

links.php
<?php
//--------------------------------------------------------------------
//-  File          : links.php
//-  Project       : FVWM Home Page
//-  Date          : Fri Mar 14 21:32:08 2003
//-  Programmer    : Uwe Pross
//--------------------------------------------------------------------

$rel_path = ".";

//--------------------------------------------------------------------
// load some global definitions
//--------------------------------------------------------------------
include_once($rel_path.'/definitions.inc'); 

//--------------------------------------------------------------------
// Site definitions
//--------------------------------------------------------------------
$title          = "FVWM - Links";
$link_name      = "Links";
$link_picture   = "pictures/icons/links";
$parent_site    = "top";
$child_sites    = array("customize_theme");
$requested_file = basename(my_get_global("PHP_SELF", "SERVER"));
$this_site      = str_replace(".php", "", "$requested_file");

//--------------------------------------------------------------------
// check if we should stop here
//--------------------------------------------------------------------
if(isset($navigation_check)) return;

//--------------------------------------------------------------------
// load the layout file
//--------------------------------------------------------------------
if(!isset($site_has_been_loaded)) {
  $site_has_been_loaded = "true";
  include_once(sec_filename("$layout_file"));
  exit();
}
?>


<?php decoration_window_start("Links"); ?>

  <h2>Other FVWM-related sites</h2>

  <p>
    Here are some other sites (besides the 
    <a href="http://www.fvwm.org/">official</a> one) that have FVWM info:
  </p>

  <ul>
      <li><?php mylink('http://fvwmwiki.xteddy.org/', 'The FVWM Wiki Pages'); ?> are
	a shared collection of FVWM related stuff.</li>
  <li><?php mylink('http://fvwmforums.org', 'The FVWM Forum'); ?> are
  a collection of user questions and answers.</li>
  </ul>


  <!-- -------------------------------------------------------------- -->
  <h2>Window Decors and Other Configuration Tips</h2>

    <ul>
<!--
        TA -  2010-08-31 -  Link dead.
       <li><a href="http://research.cmis.csiro.au/gjw/xwindows/">Glen Lee Edward's Screen
        shots and config files page</a> ("FVWM - Do it your way!") - 
        illustrates FVWM's flexibility with configuration information and, of course,
        the obligatory screenshots.</li>
-->

<!--  
        RBW - Dead.  www.oneway.com/jay is still there, but no mention of
        anything to do with FVWM.
      <li><a href="http://www.oneway.com/fvwm/">
      Jay Kuri's Fvwm Module documentation in Palm Pilot format</a></li>
-->

      <li><a href="http://ironphoenix.org/tril/fvwm/configs/">Suzanne Skinner's
      FVWM site</a> - Stylish screenshots (and config files) from the author
      of the MultiPixmap titlebar feature.
      Don't miss her thoughtful and entertaining essay
      <a href="http://ironphoenix.org/tril/fvwm/">"My Quest For The Perfect Window Manager".</a>
      <i>(The hobbit in us would like to subtitle it "There and Back Again".)</i>
      </li>

      <li><a href="http://www.twobarleycorns.net/fvwm-decors.html">Dorothy Robinson's
        FVWM 2.6 Decors</a> - an impressive collection of window decors showing off
        FVWM 2.6's visual versatility. All the config files are downloadable. There is
        also a collection of <a href="http://www.twobarleycorns.net/fvwm24-decors.html">
        decors that can be used with 2.4</a>.
      </li>
<!--
    TA:  Link dead:  2010-08-31.

      <li><a href="http://ayukawa.spb.ru/~fox/fvwm/">FVWM site from Andrew Dolgov</a> - 
        Nice screenshots, window decorations, configuration files and shell scripts.</li>
-->
      <li><a href="http://zensites.net/fvwm/guide/">FVWM Beginners Guide</a> - 
        Good to read when you start with FVWM</li>
    </ul>



  <!-- ======================================================== -->
  <h2>Themes</h2>

  <p>You say you're into desktop themes? Then one of these sites should be your
  next stop:</p>

    <ul>
      <li><a href="http://fvwm-themes.sourceforge.net/">The Official
      FVWM Themes Home Page</a> - highly recommended for novice users. FVWM Themes
      provides a powerful gui-driven theme engine and a wide variety of preconfigured
      themes so you can get started without having to learn anything about FVWM
      configuration commands. If the supplied themes don't suit you, you can create
      your own, or mix and match components from the existing ones.<br>
      The site also has links to other theme related sites.</li>

      <li><a href="http://wm-icons.sourceforge.net/">The Window Manager Icons Project</a> - 
      an efficient icon distribution to use with fvwm and fvwm-themes.</li>

<!--
        RBW - does anyone have an updated link? Can't locate this server.
        It's dead, Jim.
        A search of Colorado State's Web site doesn't turn up a Scott Scriven.
        Last known address is Anchorage, AK.  
        forum.beringsea.com/users/scott    toykeeper.com

      <li><a href="http://www.vis.colostate.edu/~scriven/fvwm.php3">Scott
      Scriven's theme page</a>.</li>
-->
      <li><a href="http://www.xwinman.org/fvwm.php">FVWM
      Decorations</a> - Some screenshots and config files</li>

    </ul>


  <!-- ======================================================== -->
  <h2>Add-ons</h2>

    <p>And here are some sites featuring programs that are designed
    to work with or enhance FVWM:</p>

    <ul>

      <li><a href="http://fvwm-ewmh.sourceforge.net">Extended WM Hints for fvwm 2.4</a> - 
        Enables integration with KDE2, GTK+2 and the future GNOME2 (this support has
        been integrated by the author into the FVWM 2.6 core)</li>

      <li><a href="http://www.blackie.dk/dotfile/">The Dotfile Generator Home Page</a> - 
      A program that helps you generate .emacs, .cshrc, .fvwmrc, etc. config files (pretty old)</li>

	  <!--
	  User lapeyre no longer exists at this site -SS (May '07)
      <li><a href="http://newton.physics.arizona.edu/~lapeyre/fvwmconf/index.html">John Lapeyre's FvwmConf</a> -
      A module for interactively configuring FVWM (old, 1997)</li>
	  -->

	  <!--
	  No mention of FvwmOsd at home.gna.org or gna.org -SS (May '07)
      <li><a href="http://home.gna.org/felidae/FvwmOsd.html">FvwmOsd</a> - display 
      an on-screen display (like the one on many TVs) on your X display</li>
	  -->

<!--
       Whatever happened to this one?  -RBW
       <li><a href="http://www-personal.umich.edu/~markcrim/tkgoodstuff/">
       The TkGoodStuff Home Page</a> - 
       A Tcl/Tk Program that performs like GoodStuff/FvwmButtons.</li> 
-->
    </ul>


  <!-- -------------------------------------------------------------- -->
  <h2>Other Window Managers</h2>

  <p>  
    Here are some window managers that were originally created from fvwm
    source, but 
    <strong>please don't ask questions about them on the FVWM lists!</strong>
    Rather, contact the parties responsible for them directly:</p>

    <ul>
<!-- 
        RBW - doesn't seem to have been updated in about 2 years (March 2000).
        Guess they're not active. Seems to be just archived at sourceforge,
        like fvwm95.
-->
      <li><a href="http://scwm.sourceforge.net/">SCWM</a>
        - The Scheme Configurable Window Manager</li>

      <li><a href="http://www.afterstep.org/">AfterStep</a>
        - A window manager that emulates the famous NEXTSTEP<sup>&reg;</sup>
        look and feel, AfterStep took up where Bo Yang's Bowman (no longer
        available) left off.</li>

      <li><a href="http://fvwm95.sourceforge.net/">FVWM-95</a>
        - An old hack of fvwm-2.0.42 that looks more like (shudder) Win95.</li>

      <li><a href="http://lesstif.org/">Lesstif's MWM</a>
        - Intended to simulate a Motif-like window manager.
        Abandoned in 1997.</li>

      <li><a href="http://xfce.org/">XFwm</a>
        - A part of the XFce desktop enviroment.</li>
    </ul>

  <p>
    Matt Chapman's excellent survey of window managers present and past
    (from the famous to the downright obscure)
    should be required reading for anyone choosing a window manager
    (or trying to write one).
  </p>

    <ul>
      <li><a href="http://www.PLiG.org/xwinman/">Window Managers for X</a><br>
    </ul>
  
  <!-- -------------------------------------------------------------- -->
  <h2>General Resources</h2>

  <p>Don't forget about Newsgroups:</p>

  <ul>
    <li><a href="news:comp.windows.x.apps">comp.windows.x.apps</a>
      - for discussing and announcing apps like FVWM.
      (Questions about FVWM are best asked by sending mail to
      <a href="mailto:fvwm@fvwm.org">fvwm@fvwm.org</a>.)</li>
    
    <li><a href="news:comp.windows.x">comp.windows.x</a>
      - for general X11 discussions and advice.</li>
  </ul>

  <!--
  And how about an online version of the ICCCM:
  -->
  <p>
    Finally, if you want to know why FVWM does some of the things it does
    (or if you just want to see if we're really behaving properly!),
    you can peruse the online version of the ICCCM:
  </p>

    <ul>
      <li><A HREF="http://tronche.com/gui/x/icccm/">Inter-Client Communication Conventions Manual</A></li>
    </ul>

  <!-- -------------------------------------------------------------- -->
  <h2>Miscellaneous</h2>

  <p>
    <ul>
      <li><a href="http://freshmeat.net/projects/fvwm/?topic_id=56">FVWM
	  freshmeat page</a> - New stable releases are announced on
	freshmeat.
    </ul>
  </p>

<?php decoration_window_end() ?>
